src/SMSClients/HubtelSMSClient.php

```diff
@@ -5,57 +5,130 @@
-        $this->client = $client;
+        $this->validateConfig($this->apiKey, $this->apiSecret);
+
+        $payload = [];
+        foreach (get_object_vars($message) as $property => $value) {
+            if (! is_null($value)) {
+                $payload[$property] = $value;
+            }
+        }
+
+        $auth = base64_encode($this->apiKey . ':' . $this->apiSecret);
+
+        return $this->client->post($this->getBaseApiURL(), [
+            'headers' => [
+                'Authorization' => 'Basic ' . $auth,
+                'Content-Type' => 'application/json',
+            ],
+            'json' => $payload,
+        ]);
     }
 
-    public function send(HubtelMessage $message)
+    /**
+     * Send SMS using GET method with credentials in query params (Quick Send)
+     *
+     * @param HubtelMessage $message
+     * @return \Psr\Http\Message\ResponseInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws InvalidConfiguration
+     */
+    protected function sendViaGet(HubtelMessage $message): \Psr\Http\Message\ResponseInterface
     {
-        return $this->client->get($this->getApiURL().$this->buildMessage($message, $this->apiKey, $this->apiSecret));
+        return $this->client->get($this->getBaseApiURL().'?'.$this->buildMessage($message, $this->apiKey, $this->apiSecret));
     }
 
-    public function getApiURL()
+    /**
+     * @inheritDoc
+     */
+    protected function getBaseApiURL(): string
     {
-        return 'https://smsc.hubtel.com/v1/messages/send?';
+        return 'https://smsc.hubtel.com/v1/messages/send';
     }
 
-    public function buildMessage(HubtelMessage $message, $apiKey, $apiSecret)
+    /**
+     * @param HubtelMessage $message
+     * @param string $apiKey
+     * @param string $apiSecret
+     * @return string
+     * @throws InvalidConfiguration
+     */
+    public function buildMessage(HubtelMessage $message, string $apiKey, string $apiSecret): string
     {
         $this->validateConfig($apiKey, $apiSecret);
 
-        $params = ['ClientId'=>$apiKey, 'ClientSecret' => $apiSecret];
+        $params = ['clientid' => $apiKey, 'clientsecret' => $apiSecret];
 
         foreach (get_object_vars($message) as $property => $value) {
             if (! is_null($value)) {
-                $params[ucfirst($property)] = $value;
+                // Convert property names to lowercase to match API requirements for GET requests
+                $params[strtolower($property)] = $value;
             }
         }
 
         return http_build_query($params);
     }
 
-    public function validateConfig($apiKey, $apiSecret)
+    /**
+     * @inheritDoc
+     */
+    protected function parseResponse(\Psr\Http\Message\ResponseInterface $response): HubtelResponse
     {
-        if (is_null($apiKey)) {
-            throw InvalidConfiguration::apiKeyNotSet();
-        }
+        $body = (string) $response->getBody();
+        $data = json_decode($body, true);
```

Comment:
Just for your information. I know I did the same in the past, but there is a flag https://www.php.net/manual/en/json.constants.php#constant.json-throw-on-error which causes `json_decode` to throw an exception on error instead of setting the global state in `json_last_error`